### PR TITLE
Docs improvement 

### PR DESCRIPTION
**Pull request by mietlinski-nc**
Restructure and add more detailed explainations in the parameters files defined in the following issues:
- https://github.com/kentrikos/template-environment-configuration/issues/2
- https://github.com/kentrikos/template-environment-configuration/issues/3

### CODE REVIEW

File: application/eu-central-1/terraform.tfvars [hunk 1]
```diff
@@ -1,44 +1,177 @@
...
+#   "arn:aws:iam::210987654321:policy/nodes.demo-test.k8s.local",
+# ]
 
 k8s_nodes_iam_policies_arns = [
   "arn:aws:iam::210987654321:policy/nodes.demo-test.k8s.local",
-] # List of existing IAM policies that will be attached to instance profile for worker nodes (EC2 instances)
-
-k8s_private_subnets = ["subnet-XXXX", "subnet-ZZZZ", "subnet-YYYY"] # List of private subnets (matching AZs) where to deploy the cluster (required if existing VPC is used)
-
-k8s_aws_ssh_keypair_name = "" # Optional name of existing SSH keypair on AWS account, to be used for cluster instances (will be generated if not specified)
```
Comment (mzawadzk) [inline application/eu-central-1/terraform.tfvars]: This variable should not be removed.

File: operations/eu-central-1/terraform.tfvars [hunk 2]
```diff
@@ -1,61 +1,223 @@
...
-
-operations_aws_account_number = "123456789012" # AWS operations account number (without hyphens)
-
-product_domain_name = "demo" # Name of product domain, will be used to create other names
-
-region = "eu-central-1" # AWS region
-
-vpc_id = "vpc-XXXXXX" # ID of existing VPC (must be pre-created)
-
-jenkins_dns_domain_hosted_zone_ID = "ABCDEFGHIJKLMN" # R53 Hosted Zone ID for domain that will be used by Jenkins master
```
Comment (radepal) [inline operations/eu-central-1/terraform.tfvars]: missing

File: operations/eu-central-1/terraform.tfvars [hunk 3]
```diff
@@ -1,61 +1,223 @@
...
-
-jenkins_dns_hostname = "jenkins" # Local part of FQDN for Jenkins master
```
Comment (radepal) [inline operations/eu-central-1/terraform.tfvars]: missing

File: application/eu-central-1/terraform.tfvars (hunk 1 above)
Comment (mietlinski-nc) [reply application/eu-central-1/terraform.tfvars]: OK, I'll add it to both tfvars files

File: operations/eu-central-1/terraform.tfvars [hunk 4]
```diff
@@ -1,61 +1,223 @@
...
 
 k8s_nodes_iam_policies_arns = [
-  "arn:aws:iam::123456789012:policy/nnodes.mpas-demo-ops.k8s.local",
+  "arn:aws:iam::123456789012:policy/nodes.maps-demo-ops.k8s.local",
   "arn:aws:iam::123456789012:policy/AssumeKopsCrossAccount",
-] # List of existing IAM policies that will be attached to instance profile for worker nodes (EC2 instances)
-
-k8s_private_subnets = ["subnet-XXXX", "subnet-ZZZZ", "subnet-YYYY"] # List of private subnets (matching AZs) where to deploy the cluster (required if existing VPC is used)
-
-k8s_aws_ssh_keypair_name = "" # Optional name of existing SSH keypair on AWS account, to be used for cluster instances (will be generated if not specified)
```
Comment (mzawadzk) [inline operations/eu-central-1/terraform.tfvars]: Missing.

File: application/eu-central-1/terraform.tfvars (hunk 1 above)
Comment (mietlinski-nc) [reply application/eu-central-1/terraform.tfvars]: Done

File: operations/eu-central-1/terraform.tfvars (hunk 2 above)
Comment (mietlinski-nc) [reply operations/eu-central-1/terraform.tfvars]: Accepted, WiP

File: operations/eu-central-1/terraform.tfvars (hunk 3 above)
Comment (mietlinski-nc) [reply operations/eu-central-1/terraform.tfvars]: Accepted, WiP

File: operations/eu-central-1/terraform.tfvars (hunk 4 above)
Comment (mietlinski-nc) [reply operations/eu-central-1/terraform.tfvars]: Added, done.

File: operations/eu-central-1/terraform.tfvars (hunk 2 above)
Comment (mietlinski-nc) [reply operations/eu-central-1/terraform.tfvars]: Done with comment

File: operations/eu-central-1/terraform.tfvars (hunk 3 above)
Comment (mietlinski-nc) [reply operations/eu-central-1/terraform.tfvars]: Added with comment in Core Infra Jenkins section

File: application/eu-central-1/terraform.tfvars [hunk 5]
```diff
@@ -1,44 +1,182 @@
...
+k8s_master_instance_type = "m4.large"
+k8s_node_instance_type = "m4.large"
+k8s_node_count = "3"
 
-k8s_private_subnets = ["subnet-XXXX", "subnet-ZZZZ", "subnet-YYYY"] # List of private subnets (matching AZs) where to deploy the cluster (required if existing VPC is used)
+# Optional name of existing SSH keypair on AWS account, to be used for cluster 
+# instances (will be generated if not specified)
 
-k8s_aws_ssh_keypair_name = "" # Optional name of existing SSH keypair on AWS account, to be used for cluster instances (will be generated if not specified)
+k8s_aws_ssh_keypair_name = ""
```
Comment (richardzimmermann) [inline application/eu-central-1/terraform.tfvars]: Please also add at least one example on how to configure the parameter
```suggestion
# k8s_aws_ssh_keypair_name = "my-keypair-name"
```

Comment (richardzimmermann) [inline application/eu-central-1/terraform.tfvars]: I'd love to see at least one example per parameter for easier handling: 

```suggestion
# k8s_aws_ssh_keypair_name = "my-keypair-name"
 
k8s_aws_ssh_keypair_name = "" 
```

File: operations/eu-central-1/env/jenkins/parameters.yaml [hunk 6]
```diff
@@ -1,14 +1,36 @@
...
+###############################################################################
+
+# This section is used by JenkinsX to import application jobs from Bitbucket
+# repository.
+# IMPORTANT NOTE - This section is mandatory, however it will be used only with
+# application deployed by JenkinsX.
+
+# Below parameters can be taken from Bitbucket settings section:
+# Repo username with assigned token to access repository
 gitUsername: ''
```
Comment (richardzimmermann) [inline operations/eu-central-1/env/jenkins/parameters.yaml]: I'd love to see at least one example per parameter for easier handling: 

```suggestion
# gitUsername: 'mygitusername'
# gitApiToken: '55b8bfdc5c74833fb125140a45a7e4786b11ba7f '

gitUsername: ''
gitApiToken: ''
```

File: operations/eu-central-1/terraform.tfvars [hunk 7]
```diff
@@ -1,61 +1,241 @@
...
+# environment_type = "test"
+# product_domain_name = "demo"
+
+environment_type = "test"
+product_domain_name = "demo"
+
+# IMPORTANT NOTE: Currently only one or three availability zones are supported.
+#
+# Parameters to configure the network for your deployment. Enter the target AWS
+# region, availability zones an existing VPC ID (for operations account only
```
Comment (mzawadzk) [inline operations/eu-central-1/terraform.tfvars]: typo, should say: "and existing vpc"

File: operations/eu-central-1/terraform.tfvars [hunk 8]
```diff
@@ -1,61 +1,241 @@
...
+# Example:
+# region = "eu-central-1"
+# azs = ["eu-central-1a", "eu-central-1b", "eu-central-1c"]
+# vpc_id = "vpc-abcdefg0123456789"
+
+region = "eu-central-1"
+azs = ["eu-central-1a", "eu-central-1b", "eu-central-1c"]
+vpc_id = "vpc-XXXXXX"
+
+# Proxy settings to use for internet access. Enter the proxy address (incl.
```
Comment (mzawadzk) [inline operations/eu-central-1/terraform.tfvars]: typo: Internet

File: operations/eu-central-1/env/jenkins/parameters.yaml [hunk 9]
```diff
@@ -1,14 +1,36 @@
...
+# defaultAdminPassword: 'qwerty'
 defaultAdminPassword: 'qwerty'
 
-# DNS configuration:
-## R53 Hosted Zone ID for domain that will be used by jenkins-x:
-jxDomainHostedZoneID: 'AAAAAAAAAAAAAA'
-## Prefix (string) for wildcard alias that will be created (*.jxDomainAliasPrefix.DOMAIN)
-jxDomainAliasPrefix: 'abc'
+# jxHostedZoneID is an AWS Route53 parameter to store Domain for Jx
+jxDomainHostedZoneID: ''
```
Comment (richardzimmermann) [inline operations/eu-central-1/env/jenkins/parameters.yaml]: I'd love to see at least one example per parameter for easier handling: 

```suggestion
# Example:
# jxDomainHostedZoneID: 'Z9NMCFPV6TPZZR'

jxDomainHostedZoneID: ''
```

File: application/eu-central-1/terraform.tfvars (hunk 5 above)
Comment (mietlinski-nc) [reply application/eu-central-1/terraform.tfvars]: Accepted, WiP

File: operations/eu-central-1/terraform.tfvars [hunk 10]
```diff
@@ -1,61 +1,241 @@
...
+jenkins_dns_hostname = "jenkins" 
+
+###############################################################################
+# Kubernetes Cluster Configuration in Operations account
+###############################################################################
+
+# Domain name of Kubernetes cluster (currently only k8s.local is supported)
 
-auto_IAM_mode = false # Create IAM Policies in AWS (default false)
+k8s_cluster_name_postfix = "k8s.local"
```
Comment (richardzimmermann) [inline operations/eu-central-1/terraform.tfvars]: I'd love to see at least one example per parameter for easier handling:

```suggestion
# Example (and only allowed value at this moment):
# k8s_cluster_name_postfix = "k8s.local"

k8s_cluster_name_postfix = "k8s.local"
```

File: operations/eu-central-1/terraform.tfvars [hunk 11]
```diff
@@ -1,61 +1,241 @@
...
+# k8s_node_count = "3"
+
+k8s_master_instance_type = "m4.large"
+k8s_node_instance_type = "m4.large"
+k8s_node_count = "3"
+
+# Optional name of existing SSH keypair on AWS account, to be used for cluster 
+# instances (will be generated if not specified)
+
+k8s_aws_ssh_keypair_name = ""
```
Comment (richardzimmermann) [inline operations/eu-central-1/terraform.tfvars]: I'd love to see at least one example per parameter for easier handling: 

```suggestion
# Example:
# k8s_aws_ssh_keypair_name = "my-keypair-name"
 
k8s_aws_ssh_keypair_name = "" 
```

File: operations/eu-central-1/terraform.tfvars (hunk 7 above)
Comment (mietlinski-nc) [reply operations/eu-central-1/terraform.tfvars]: Corrected

File: operations/eu-central-1/terraform.tfvars (hunk 8 above)
Comment (mietlinski-nc) [reply operations/eu-central-1/terraform.tfvars]: Corrected

**Pull request merged**